### ISSUE TITLE
Evaluate optimizer

### ISSUE THREAD

**Issue by George3d6** [mindsdb/lightwood#208]
The current optimizer is a franken monster mix that's basically lookahead + radam, so it's basically an optimizer + scheduler + regularization mechanism all in one.

I'm not sure this is ideal, at the time I benchmarked it and it seemed nice (reached slightly higher accuracies and was faster), and fastai swears by Radam, but honestly that's not rigorous enough.

We should validate the optimizer once we have the benchmarks and replace it with something that results in very stable loss progression and is true and tested (e.g. SGD or SGD + very simple linear-increase then flat scheduler) if it doesn't prove very useful.

@torrmal @paxcema  if you have opinions on this do tell.

**Comment by George3d6** [comment]
Ok, so, I've done a bit of experimenting as part of playing around with other features and comparing rangers (RAdam + Lookaehad) with SGD and AdamW.

What I will note is something I forgot:

This optimizer is actually pretty good at having gradient never hit `nan` given a small enough learning rate.

On certain networks AdamW and SGD work with a lr in the 0.01 - 0.1 range, on other, they only work if the lr is > 1e-4 (at least for starters, and in these cases we'd have to apply a scheduler to increase it)

On the other hand, if the lr is small but not unreasonably so (0.001) it seems almost impossible for gradients to go to `nan` with ranger (not sure if this is more due to AdamR or Lookahead). To add to that, AdamR does seem to handle the lr scheduling such that training ends up being faster in some cases than with SGD, even though the optimizer `step` takes significantly longer.

Which is not to say we should keep this optimizer, but I think I am back to being on the fence about it. We should wait for benchmarks and only then change it.

**Comment by George3d6** [comment]
We more or less switched to optuna I guess, for lightgbm, it's really easy to use which is nice, we might try using it in the future for the nn mixers and the encoders, but leaving this aside as a "discusion" for now.
